arxiv/identifier/iteration.py

```diff
@@ -0,0 +1,45 @@
+from typing import Optional, List
+
+from sqlalchemy import or_
+
+from ..db import get_db
+from ..db.models import Metadata
+from . import Identifier
+from ..taxonomy.category import Category
+from ..document.version import SOURCE_FORMAT
+
+
+class arXivIDIterator:
+
+    def __init__ (self,
+                  start_yymm: str,
+                  end_yymm: str,
+                  categories: Optional[List[Category]] = None,
+                  source_formats: Optional[List[SOURCE_FORMAT]] = None,
+                  only_latest_version: bool = False):
+        
+        with get_db() as session:
+            self.query = session.query(Metadata.paper_id, Metadata.version) \
+                .filter(Metadata.paper_id >= start_yymm) \
+                .filter(Metadata.paper_id < f'{end_yymm}.999999')
+            if categories:
+                self.query = self.query.filter(or_(*[Metadata.abs_categories.like(f'%{category.id}%') for category in categories]))
```

Comment:
possible hiccup here due to some entries not including the most modern/relevant name of their category. Technically this code is fine, but to find every paper whatever calls this would have to include all possible category names (both versions of aliases and subsumed arxivs)

Reply:
Hmm yeah I wrote this piece of code sort of carelessly, and I don't know what it would be useful for. I'd like to search/sort by announce date but that doesn't seem easy with the columns available to me. I may just delete it and write again when I start work on converting the full corpus to HTML

Reply:
ive spent way too much time figuring how to search for all papers of a given demographic. LMK when you work on that